### ISSUE TITLE
Internal Server Error 500

### ISSUE THREAD

**Issue by nzkller** [janeczku/calibre-web#541]
Hello guys,

I've install the application and after installing all the dependencies, even the optional ones and the kindlegen bin on the vendor folder.

When trying to connect to the application I get 

### **Internal Server Error**
The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.

On the console I can see the following errors
`
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/root/calibre/calibre-web/cps/web.py", line 482, in inner
    return f(*args, **kwargs)
  File "/root/calibre/calibre-web/cps/web.py", line 2491, in basic_configuration
    return configuration_helper(1)
  File "/root/calibre/calibre-web/cps/web.py", line 2500, in configuration_helper
    gdriveError = _('Import of optional GDrive requirements missing')
  File "/root/calibre/calibre-web/vendor/flask_babel/__init__.py", line 555, in gettext
    s = t.ugettext(string)
  File "/usr/lib64/python2.7/gettext.py", line 402, in ugettext
    tmsg = self._catalog.get(message, missing)
AttributeError: 'Translations' object has no attribute '_catalog'
ERROR:tornado.access:500 GET /config (xxx.xxx.xxx.xxx) 12.88ms
`

For a moment I thought it was my apache config but I went straight to 127.0.0.1:8083 and I get the same errors.

Nevertheless, I turned on the debug for apache and I can't see any errors 
`[Fri Jul 13 12:37:18.146122 2018] [ssl:info] [pid 24343] [client xxx.xxx.xxx.xxx:58531] AH01964: Connection to child 1 established (server webserver.com:443)
[Fri Jul 13 12:37:18.146372 2018] [ssl:debug] [pid 24343] ssl_engine_kernel.c(1890): [client xxx.xxx.xxx.xxx:58531] AH02043: SSL virtual host for servername webserver.com found
[Fri Jul 13 12:37:18.146424 2018] [ssl:debug] [pid 24343] ssl_util_stapling.c(609): AH01951: stapling_cb: OCSP Stapling callback called
[Fri Jul 13 12:37:18.146444 2018] [ssl:debug] [pid 24343] ssl_util_stapling.c(617): AH01952: stapling_cb: retrieved cached certificate data
[Fri Jul 13 12:37:18.146501 2018] [socache_shmcb:debug] [pid 24343] mod_socache_shmcb.c(522): AH00835: socache_shmcb_retrieve (0x8d -> subcache 1)
[Fri Jul 13 12:37:18.146515 2018] [socache_shmcb:debug] [pid 24343] mod_socache_shmcb.c(845): AH00849: match at idx=0, data=0
[Fri Jul 13 12:37:18.146522 2018] [socache_shmcb:debug] [pid 24343] mod_socache_shmcb.c(532): AH00836: leaving socache_shmcb_retrieve successfully
[Fri Jul 13 12:37:18.146536 2018] [ssl:debug] [pid 24343] ssl_util_stapling.c(302): AH01933: stapling_get_cached_response: cache hit
[Fri Jul 13 12:37:18.146540 2018] [ssl:debug] [pid 24343] ssl_util_stapling.c(631): AH01953: stapling_cb: retrieved cached response
[Fri Jul 13 12:37:18.146591 2018] [ssl:debug] [pid 24343] ssl_util_stapling.c(675): AH01956: stapling_cb: setting response
[Fri Jul 13 12:37:18.174671 2018] [ssl:debug] [pid 24343] ssl_engine_kernel.c(1823): [client xxx.xxx.xxx.xxx:58531] AH02041: Protocol: TLSv1.2, Cipher: ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
[Fri Jul 13 12:37:18.178133 2018] [ssl:debug] [pid 24343] ssl_engine_kernel.c(225): [client xxx.xxx.xxx.xxx:58531] AH02034: Initial (No.1) HTTPS request received for child 1 (server webserver.com:443)
[Fri Jul 13 12:37:18.178278 2018] [authz_core:debug] [pid 24343] mod_authz_core.c(835): [client xxx.xxx.xxx.xxx:58531] AH01628: authorization result: granted (no directives)
[Fri Jul 13 12:37:18.178354 2018] [proxy:debug] [pid 24343] mod_proxy.c(1123): [client xxx.xxx.xxx.xxx:58531] AH01143: Running scheme http handler (attempt 0)
[Fri Jul 13 12:37:18.178366 2018] [proxy_ajp:debug] [pid 24343] mod_proxy_ajp.c(722): [client xxx.xxx.xxx.xxx:58531] AH00894: declining URL http://webserver.com:8083//config
[Fri Jul 13 12:37:18.178374 2018] [proxy_fcgi:debug] [pid 24343] mod_proxy_fcgi.c(972): [client xxx.xxx.xxx.xxx:58531] AH01076: url: http://webserver.com:8083//config proxyname: (null) proxyport: 0
[Fri Jul 13 12:37:18.178400 2018] [proxy_fcgi:debug] [pid 24343] mod_proxy_fcgi.c(975): [client xxx.xxx.xxx.xxx:58531] AH01077: declining URL http://webserver.com:8083//config
[Fri Jul 13 12:37:18.178440 2018] [proxy:debug] [pid 24343] proxy_util.c(2203): AH00942: HTTP: has acquired connection for (webserver.com)
[Fri Jul 13 12:37:18.178461 2018] [proxy:debug] [pid 24343] proxy_util.c(2256): [client xxx.xxx.xxx.xxx:58531] AH00944: connecting http://webserver.com:8083//config to webserver.com:8083
[Fri Jul 13 12:37:18.179276 2018] [proxy:debug] [pid 24343] proxy_util.c(2426): [client xxx.xxx.xxx.xxx:58531] AH00947: connected //config to webserver.com:8083
[Fri Jul 13 12:37:18.179400 2018] [proxy:debug] [pid 24343] proxy_util.c(2802): AH02824: HTTP: connection established with [xxxx:xxxx:xxxx:xxxx::xxxx]:8083 (webserver.com) ---> In here I see the IPv6 address of the server
[Fri Jul 13 12:37:18.179451 2018] [proxy:debug] [pid 24343] proxy_util.c(2942): AH00962: HTTP: connection complete to [xxxx:xxxx:xxxx:xxxx::xxxx]:8083 (webserver.com) ---> In here I see the IPv6 address of the server
[Fri Jul 13 12:37:18.189050 2018] [proxy:debug] [pid 24343] proxy_util.c(2218): AH00943: http: has released connection for (webserver.com)
[Fri Jul 13 12:37:18.189079 2018] [headers:debug] [pid 24343] mod_headers.c(823): AH01502: headers: ap_headers_output_filter()
[Fri Jul 13 12:37:18.189202 2018] [ssl:debug] [pid 24343] ssl_engine_io.c(993): [client xxx.xxx.xxx.xxx:58531] AH02001: Connection closed to child 1 with standard shutdown (server webserver.com:443)`

Thanks in advance for the help and best regards,

Christian Dassy

**Comment by aroundmyroom** [comment]
there seems something going wrong with:
"configuration_helper gdriveError = _('Import of optional GDrive requirements missing') File "

**Comment by OzzieIsaacs** [comment]
The error hints towards some problem with Babel, flask-babel library. May I ask which versions of both libraries are installed?

**Comment by OzzieIsaacs** [comment]
Have you installed it on a „normal“ pc, or is it some sort of NAS, raspberry pi or something else?

**Comment by nzkller** [comment]
Hi @OzzieIsaacs ,

Thank you very much you pinpointed the issue and and directed me to the solution.

It was a problem with Babel & flask-babel libraries. Basically they were not installed. When I did **pip list** there were not there 😞. I don't know how I missed it 🤣. Thank you again for your input.

I installed in a Centos 7 Server I have, So i can use my calibre library from anywhere when I'm at work, etc.

Thank you all and have a great weekend!!!
